### ISSUE TITLE
How to migrate legacy view bake helpers (and templates)?

### ISSUE THREAD

**Issue by geoidesic** [cakephp/bake#703]
I have a bake helper like so:
```
<?php
namespace QuasarAdmin\View\Helper;

use Cake\Datasource\ConnectionManager;
use Bake\View\Helper\BakeHelper as BaseHelper;

/**
 * Bake helper
 */
class BakeHelper extends BaseHelper
{
    public function checkTableExistsByAlias($alias) {
        $exists = false;
        $connection = ConnectionManager::get('default');
        $tables = $connection->execute('show tables;')->fetchAll('assoc');
        foreach (array_values($tables) as $key => $tableName) {
            if ($alias === array_values($tableName)[0]) {
                $exists = true;
            }
        }
        return $exists;
    }
}
```
However the new `twig-view` is not picking up this extension. How can I make it do so?

**Comment by geoidesic** [comment]
Incidentally... I've been using CakePHP 4.1.1 for months without incident, on an upgraded project. This only occurs on a freshly created project. It seems that on upgrades, `twig-view` isn't upgraded and so my bake templates and helper still works. However on the fresh install its broken.

I also tried removing `cakephp/twig-view` and requiring `cakephp/twig-view-legacy` or `wyrihaximus/twig-view` but it wouldn't let me do that.

**Comment by alexdd55** [comment]
> Also it looks like my bake templates from that plugin are no longer being used either. I.e. those in
> `\plugins\QuasarAdmin\templates\bake`

Same issue here. To solve this temporarily i had to copy my templates into the vendor folder then baking works as usual.

**Comment by othercorey** [comment]
Can you give an example of using your helper? In twig-view, you access helper through `helper_name` but they are not "auto loaded".

**Comment by ADmad** [comment]
bake now use `TwigView` which requires helpers to be explicitly loaded to be available as twig methods in the twig templates.

There `TemplateRender` used by bake commands will always uses bake's https://github.com/cakephp/bake/blob/master/src/Utility/TemplateRenderer.php#L72.

So if you are extending the `BakeHelper` you'll have to also configure `TemplateRender` to use your extended helper. It's better to just create another helper and load that.

**Comment by ADmad** [comment]
Regarding templates if you want to use your plugin's templates then use bake's [theme](https://book.cakephp.org/bake/2/en/development.html#creating-a-bake-theme) feature.

**Comment by othercorey** [comment]
@geoidesic Can you post an example of the template you're porting? If not, we can't do anything.

**Comment by geoidesic** [comment]
@othercorey so... it's a plugin with many templates. I'm not sure what to post?

**Comment by geoidesic** [comment]
This is what the folder structure of the plugin looks like:

```
/Users/me/plugins/QuasarAdmin/templates/bake
/Users/me/plugins/QuasarAdmin/templates/bake/Controller
/Users/me/plugins/QuasarAdmin/templates/bake/Controller/api.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Controller/controller.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller/add.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller/delete.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller/edit.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller/index.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element/Controller/view.twig
/Users/me/plugins/QuasarAdmin/templates/bake/element/form.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Model
/Users/me/plugins/QuasarAdmin/templates/bake/Model/baked.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Model/extension.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Model/main.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Template
/Users/me/plugins/QuasarAdmin/templates/bake/Template/add.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Template/edit.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Template/index.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Template/login.twig
/Users/me/plugins/QuasarAdmin/templates/bake/Template/view.twig
```

**Comment by othercorey** [comment]
Well, you seem to have trouble updating the templates to the twig-view syntax. How can we guess what issue you have?

**Comment by geoidesic** [comment]
The way I bake this is with the following command: `bin/cake bake all --everything --theme QuasarAdmin`

**Comment by geoidesic** [comment]
> Well, you seem to have trouble updating the templates to the twig-view syntax. How can we guess what issue you have?

I don't understand what that means. I haven't tried updating any templates. Wasn't aware that they need updating. All I know is that my plugin doesn't work on a new installation of 4.1.x

**Comment by geoidesic** [comment]
I'm happy to help :) I just don't understand what or how to help. So if you explain what you need from me, I'll do it.

**Comment by othercorey** [comment]
> @geoidesic Can you post an example of the template you're porting? If not, we can't do anything.

Show us an example of the template that has an issue with the helper or other rendering issues. I don't understand what point you're making. You didn't respond to @ADmad about extending the helper earlier.

**Comment by geoidesic** [comment]
tbh I didn't understand his comment... it went over my head. I'll try set it up again so that I have a repo where the error is occurring and then I'll try provide as much info as I can.

**Comment by geoidesic** [comment]
Ok so I've got it set up and I've tried running bake again. Several things are going wrong. Firstly it's baking files to the wrong folders. I've overridden `Command/ControllerCommand.php` so that I can control which folder it writes controllers to. I've done something similar for the `ModelCommand`. These folder output overrides are respected in `wyrihaximus/twig-view` but not in `cakephp/twig-view`.

Here's my `ControllerCommand` extension as an example:
```
<?php
declare (strict_types = 1);

/**
 * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
 * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
 *
 * Licensed under The MIT License
 * For full copyright and license information, please see the LICENSE.txt
 * Redistributions of files must retain the above copyright notice.
 *
 * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
 * @link          http://cakephp.org CakePHP(tm) Project
 * @since         0.1.0
 * @license       http://www.opensource.org/licenses/mit-license.php MIT License
 */

namespace QuasarAdmin\Command;

use Bake\Command\ControllerCommand as BaseControllerCommand;
use Bake\Utility\TemplateRenderer;
use Cake\Console\Arguments;
use Cake\Console\ConsoleIo;
use Cake\Core\Configure;
use Cake\ORM\TableRegistry;

/**
 * Task class for creating and updating controller files.
 */
class ControllerCommand extends BaseControllerCommand
{
    /**
     * Path fragment for generated code.
     *
     * @var string
     */
    public $pathFragment = 'Controller/Baked/';

    public function getControllerContext(string $controllerName, Arguments $args, ConsoleIo $io): array
    {
        $io->quiet(sprintf('Baking controller class for %s...', $controllerName));

        $actions = [];
        if (!$args->getOption('no-actions') && !$args->getOption('actions')) {
            $actions = ['index', 'view', 'add', 'edit', 'delete'];
        }
        if ($args->getOption('actions')) {
            $actions = array_map('trim', explode(',', $args->getOption('actions')));
            $actions = array_filter($actions);
        }

        $helpers = $this->getHelpers($args);
        $components = $this->getComponents($args);

        $prefix = $this->getPrefix($args);
        if ($prefix) {
            $prefix = '\\' . str_replace('/', '\\', $prefix);
        }

        $namespace = Configure::read('App.namespace');
        if ($this->plugin) {
            $namespace = $this->_pluginNamespace($this->plugin);
        }

        $currentModelName = $controllerName;
        $plugin = $this->plugin;
        if ($plugin) {
            $plugin .= '.';
        }

        if (TableRegistry::getTableLocator()->exists($plugin . $currentModelName)) {
            $modelObj = TableRegistry::getTableLocator()->get($plugin . $currentModelName);
        } else {
            $modelObj = TableRegistry::getTableLocator()->get($plugin . $currentModelName, [
                'connectionName' => $this->connection,
            ]);
        }

        $pluralName = $this->_variableName($currentModelName);
        $singularName = $this->_singularName($currentModelName);
        $singularHumanName = $this->_singularHumanName($controllerName);
        $pluralHumanName = $this->_variableName($controllerName);

        $defaultModel = sprintf('%s\Model\Table\%sTable', $namespace, $controllerName);
        if (!class_exists($defaultModel)) {
            $defaultModel = null;
        }
        $entityClassName = $this->_entityName($modelObj->getAlias());

        $data = compact(
            'actions',
            'components',
            'currentModelName',
            'defaultModel',
            'entityClassName',
            'helpers',
            'modelObj',
            'namespace',
            'plugin',
            'pluralHumanName',
            'pluralName',
            'prefix',
            'singularHumanName',
            'singularName'
        );
        $data['name'] = $controllerName;
        return $data;
    }
    /**
     * Assembles and writes a Controller file
     *
     * @param string $controllerName Controller name already pluralized and correctly cased.
     * @param \Cake\Console\Arguments $args The console arguments
     * @param \Cake\Console\ConsoleIo $io The console io
     * @return void
     */
    public function bake(string $controllerName, Arguments $args, ConsoleIo $io): void
    {
        $data = $this->getControllerContext($controllerName, $args, $io);
        $data['crudPluginLoaded'] = \Cake\Core\Plugin::isLoaded('Crud');
        $this->bakeController($controllerName, $data, $args, $io);
        $this->bakeTest($controllerName, $args, $io);
        $options = ['template' => 'api', 'folder' => 'Api', 'onlyIfNotExists' => true];
        $this->bakeControllerToFolder($controllerName, $data, $args, $io, $options);
    }

    /**
     * Writes additionally to a folder of choice
     *
     * @param string $controllerName The name of the controller.
     * @param array $data The data to turn into code.
     * @return string The generated controller file.
     */
    public function bakeControllerToFolder(string $controllerName, array $data, Arguments $args, ConsoleIo $io, array $options): void
    {
        $data += [
            'name' => null,
            'namespace' => null,
            'prefix' => null,
            'actions' => null,
            'helpers' => null,
            'components' => null,
            'plugin' => null,
            'pluginPath' => null,
        ];

        $renderer = new TemplateRenderer($this->theme);
        $renderer->set($data);

        $contents = $renderer->generate('Controller/' . $options['template']);

        $path = $this->getPath($args);
        $filename = $path . $controllerName . 'Controller.php';
        $filename = str_replace('Baked', $options['folder'], $filename);

        if ($options['onlyIfNotExists'] && !file_exists($filename)) {
            $io->createFile($filename, $contents);
        }
    }
}
```

**Comment by geoidesic** [comment]
Secondly I'm getting errors and warnings during the bake. The first is this:
```
Deprecated Error: `element` tag is deprecated. Use `element()` function instead. - /Users/noeldacosta/repo/test-bake-twig/vendor/twig/twig/src/Parser.php, line: 182
 You can disable deprecation warnings by setting `Error.errorLevel` to `E_ALL & ~E_USER_DEPRECATED` in your config/app.php.
In [/Users/noeldacosta/repo/test-bake-twig/vendor/cakephp/cakephp/src/Core/functions.php, line 305]

2020-08-19 17:00:57 Notice: Deprecated (16384): `element` tag is deprecated. Use `element()` function instead. - /Users/noeldacosta/repo/test-bake-twig/vendor/twig/twig/src/Parser.php,
line: 182
 You can disable deprecation warnings by setting `Error.errorLevel` to `E_ALL & ~E_USER_DEPRECATED` in your config/app.php. in [/Users/noeldacosta/repo/test-bake-twig/vendor/cakephp/cakephp/src/Core/functions.php, line 305]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE/src/Error/BaseErrorHandler.php, line 188
deprecationWarning - CORE/src/Core/functions.php, line 305
Cake\TwigView\Twig\TokenParser\ElementParser::parse() - ROOT/vendor/cakephp/twig-view/src/Twig/TokenParser/ElementParser.php, line 44
Twig\Parser::subparse() - ROOT/vendor/twig/twig/src/Parser.php, line 182
Twig\TokenParser\ForTokenParser::parse() - ROOT/vendor/twig/twig/src/TokenParser/ForTokenParser.php, line 40
Twig\Parser::subparse() - ROOT/vendor/twig/twig/src/Parser.php, line 182
Twig\Parser::parse() - ROOT/vendor/twig/twig/src/Parser.php, line 95
Twig\Environment::parse() - ROOT/vendor/twig/twig/src/Environment.php, line 479
Twig\Environment::compileSource() - ROOT/vendor/twig/twig/src/Environment.php, line 507
Twig\Environment::loadTemplate() - ROOT/vendor/twig/twig/src/Environment.php, line 348
Twig\Environment::load() - ROOT/vendor/twig/twig/src/Environment.php, line 309
Cake\TwigView\View\TwigView::_evaluate() - ROOT/vendor/cakephp/twig-view/src/View/TwigView.php, line 300
Cake\View\View::_render() - CORE/src/View/View.php, line 1134
Bake\View\BakeView::render() - ROOT/vendor/cakephp/bake/src/View/BakeView.php, line 96
Bake\Utility\TemplateRenderer::generate() - ROOT/vendor/cakephp/bake/src/Utility/TemplateRenderer.php, line 101
Bake\Command\ControllerCommand::bakeController() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 183
Bake\Command\ControllerCommand::bake() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 154
Bake\Command\ControllerCommand::execute() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 65
Bake\Command\AllCommand::execute() - ROOT/vendor/cakephp/bake/src/Command/AllCommand.php, line 102
Cake\Console\BaseCommand::run() - CORE/src/Console/BaseCommand.php, line 179
Cake\Console\CommandRunner::runCommand() - CORE/src/Console/CommandRunner.php, line 336
Cake\Console\CommandRunner::run() - CORE/src/Console/CommandRunner.php, line 171
[main] - ROOT/bin/cake.php, line 12
```

**Comment by geoidesic** [comment]
And the second is this: 
```
Warning Error: Method Bake\View\Helper\BakeHelper::checkTableExistsByAlias does not exist
In [/Users/noeldacosta/repo/test-bake-twig/vendor/cakephp/cakephp/src/View/Helper.php, line 102]

2020-08-19 17:00:57 Warning: Warning (512): Method Bake\View\Helper\BakeHelper::checkTableExistsByAlias does not exist in [/Users/noeldacosta/repo/test-bake-twig/vendor/cakephp/cakephp/src/View/Helper.php, line 102]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE/src/Error/BaseErrorHandler.php, line 188
Cake\View\Helper::__call() - CORE/src/View/Helper.php, line 102
twig_get_attribute - ROOT/vendor/twig/twig/src/Extension/CoreExtension.php, line 1533
__TwigTemplate_ed84ad30349ec07c79fbe715600209008cb86645ed5342d93ee81016d4aef68e::doDisplay() - ROOT/vendor/twig/twig/src/Environment.php(358) : eval()'d code, line 147
Twig\Template::displayWithErrorHandling() - ROOT/vendor/twig/twig/src/Template.php, line 394
Twig\Template::display() - ROOT/vendor/twig/twig/src/Template.php, line 367
Twig\Template::render() - ROOT/vendor/twig/twig/src/Template.php, line 379
Twig\TemplateWrapper::render() - ROOT/vendor/twig/twig/src/TemplateWrapper.php, line 40
Cake\TwigView\View\TwigView::_evaluate() - ROOT/vendor/cakephp/twig-view/src/View/TwigView.php, line 300
Cake\View\View::_render() - CORE/src/View/View.php, line 1134
Cake\View\View::_renderElement() - CORE/src/View/View.php, line 1668
Cake\View\View::element() - CORE/src/View/View.php, line 655
__TwigTemplate_f53a6828a89d12012a16f50d64ff79254858df59fe1af2f2c39df6fb78d6e480::doDisplay() - ROOT/vendor/twig/twig/src/Environment.php(358) : eval()'d code, line 161
Twig\Template::displayWithErrorHandling() - ROOT/vendor/twig/twig/src/Template.php, line 394
Twig\Template::display() - ROOT/vendor/twig/twig/src/Template.php, line 367
Twig\Template::render() - ROOT/vendor/twig/twig/src/Template.php, line 379
Twig\TemplateWrapper::render() - ROOT/vendor/twig/twig/src/TemplateWrapper.php, line 40
Cake\TwigView\View\TwigView::_evaluate() - ROOT/vendor/cakephp/twig-view/src/View/TwigView.php, line 300
Cake\View\View::_render() - CORE/src/View/View.php, line 1134
Bake\View\BakeView::render() - ROOT/vendor/cakephp/bake/src/View/BakeView.php, line 96
Bake\Utility\TemplateRenderer::generate() - ROOT/vendor/cakephp/bake/src/Utility/TemplateRenderer.php, line 101
Bake\Command\ControllerCommand::bakeController() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 183
Bake\Command\ControllerCommand::bake() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 154
Bake\Command\ControllerCommand::execute() - ROOT/vendor/cakephp/bake/src/Command/ControllerCommand.php, line 65
Bake\Command\AllCommand::execute() - ROOT/vendor/cakephp/bake/src/Command/AllCommand.php, line 102
Cake\Console\BaseCommand::run() - CORE/src/Console/BaseCommand.php, line 179
Cake\Console\CommandRunner::runCommand() - CORE/src/Console/CommandRunner.php, line 336
Cake\Console\CommandRunner::run() - CORE/src/Console/CommandRunner.php, line 171
[main] - ROOT/bin/cake.php, line 12
```

**Comment by geoidesic** [comment]
My `BakeHelper` is stored in `/plugins/QuasarAdmin/src/View/Helper/BakerHelper.php` and looks like this:
```
<?php
namespace QuasarAdmin\View\Helper;

use Cake\Datasource\ConnectionManager;
use Bake\View\Helper\BakeHelper as BaseHelper;

/**
 * Bake helper
 */
class BakeHelper extends BaseHelper
{
    public function checkTableExistsByAlias($alias) {
        $exists = false;
        $connection = ConnectionManager::get('default');
        $tables = $connection->execute('show tables;')->fetchAll('assoc');
        foreach (array_values($tables) as $key => $tableName) {
            if ($alias === array_values($tableName)[0]) {
                $exists = true;
            }
        }
        return $exists;
    }
}

```

**Comment by geoidesic** [comment]
Would it help if I published the `QuasarAdmin` plugin to GitHub?

**Comment by geoidesic** [comment]
I'm not sure if it's a template issue or if it's a Command issue or something else... or maybe a combination.

**Comment by geoidesic** [comment]
I'm also getting this error, which I think is as a result of the output folders not being respected:
```
Fatal error: Cannot declare class App\Controller\Baked\AppointmentsController, because the name is already in use in /Users/noeldacosta/repo/test-bake-twig/src/Controller/AppointmentsController.php on line 13
```

**Comment by geoidesic** [comment]
Apologies if anything I've said is confusing I'm having a hard time understanding what's happened. There's little documentation available about the change from `wyrihaximus/twig-view` to `cakephp/twig-view` as far as I can tell. So I'm just not sure where to start. I know Admad was trying to help and probably what he's said is right but I just am not able to grasp it. I read his words and they individually have meaning but as a paragraph it's just gobbledygook to me :) Can't figure out what he means.

I.e.
1. `bake now use TwigView`... I thought it had always used TwigView?
1. ` which requires helpers to be explicitly loaded to be available as twig methods in the twig templates.`... ok but I don't know in practical terms what that means – like:
1.1 what is a `twig method` 
1.1 and how would that relate to my BakeHelper?
1.1 and how would I explicitly load the BakeHelper?
1. `There TemplateRender used by bake commands will always uses bake's https://github.com/cakephp/bake/blob/master/src/Utility/TemplateRenderer.php#L72.` I have no idea what this means or what I should be taking away from this or why he has pointed this out.
1. `So if you are extending the BakeHelper you'll have to also configure TemplateRender to use your extended helper.` - ok, I'm not sure how one would `configure TemplateRender` to make it use my extended helper.
1. ` It's better to just create another helper and load that.` - another helper? Ok, so that bit makes sense. I can make a new helper containing that method and then somehow get that loaded in the template

**Comment by geoidesic** [comment]
So the template that is using that BakeHelper is `plugins/QuasarAdmin/templates/bake/element/Controller/edit.twig`, which looks like this:

```
{#
/**
 * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
 * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
 *
 * Licensed under The MIT License
 * For full copyright and license information, please see the LICENSE.txt
 * Redistributions of files must retain the above copyright notice.
 *
 * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
 * @link          http://cakephp.org CakePHP(tm) Project
 * @since         2.0.0
 * @license       http://www.opensource.org/licenses/mit-license.php MIT License
 */
#}
{% set belongsTo = Bake.aliasExtractor(modelObj, 'BelongsTo') %}
{% set belongsToMany = Bake.aliasExtractor(modelObj, 'belongsToMany') %}
    {% set compact = [] %}
/**
     * Edit method
     *
     * @param string|null $id {{ singularHumanName }} id.
     * @return \Cake\Http\Response|null Redirects on successful edit, renders view otherwise.
     * @throws \Cake\Datasource\Exception\RecordNotFoundException When record not found.
     */
    public function edit($id = null)
    {
        $this->id = $id;
{% if not crudPluginLoaded %}
        ${{ singularName }} = $this->{{ currentModelName }}->get($id, [
            'contain' => [{{ Bake.stringifyList(belongsToMany, {'indent': false})|raw }}]
        ]);
        if ($this->request->is(['patch', 'post', 'put'])) {
            ${{ singularName }} = $this->{{ currentModelName }}->patchEntity(${{ singularName }}, $this->request->getData());
            if ($this->{{ currentModelName }}->save(${{ singularName }})) {
                $this->Flash->success(__('The {{ singularHumanName|lower }} has been saved.'));

                return $this->redirect(['action' => 'index']);
            }
            $this->Flash->error(__('The {{ singularHumanName|lower }} could not be saved. Please, try again.'));
        }
{% endif %} 
{% if crudPluginLoaded %}
        $this->Crud->on('beforeRender', [$this, '_beforeRenderEdit']);
        return $this->Crud->execute('edit',['id' => $id]);
{% else %} 
        $this->set(compact('{{singularName}}'));
        $this->_beforeRenderEdit();
{% endif %} 
    }
/**
     * Changes the render path to the Extension
     */
    public function _beforeRenderEdit() {

        $id = $this->id;
        $this->viewBuilder()
            ->setTemplatePath('Extension/{{ currentModelName }}');
{% for assoc in belongsTo|merge(belongsToMany) %}
    {%- set otherName = Bake.getAssociatedTableAlias(modelObj, assoc) %}
    {%- set otherPlural = otherName|variable %}
    {% if Bake.checkTableExistsByAlias(otherPlural) %}
    ${{ otherPlural }} = $this->{{ currentModelName }}->{{ otherName }}->find('list',['limit' => 200]);
    {% else %}
    ${{ otherPlural }} = $this->{{ currentModelName }}->find(
            'list',
            [
                'limit' => 200,
                'conditions' => [
                    '{{ currentModelName }}.id !=' => $id
                ],
                'keyField' => 'id',
                'valueField' => 'name'
            ]
        );
    {% endif %}
    {{- "\n" }}
    {%- set compact = compact|merge(["'#{otherPlural}'"]) %}
{% endfor %}
{% if compact %}
        $this->set(compact({{ compact|join(', ')|raw }}));
{% endif %}
    }
```

**Comment by othercorey** [comment]
There are some parts of your implementation we don't understand. Are you overriding `bake all` command somewhere to run your custom command?

We aren't sure how your custom bake helper is being loaded.

**Comment by ADmad** [comment]
> So if you are extending the BakeHelper you'll have to also configure TemplateRender to use your extended helper. - ok, I'm not sure how one would configure TemplateRender to make it use my extended helper.

https://github.com/cakephp/bake/pull/706

**Comment by othercorey** [comment]
@geoidesic I am looking at this again. I want to check the issue is still that the custom QuasarAdmin.Bake helper seem to not be used? We talked a lot about loading and twig porting.

**Comment by othercorey** [comment]
I added a unit test for plugin helpers based on your templates.

https://github.com/cakephp/twig-view/pull/60

**Comment by othercorey** [comment]
I don't see anything wrong in your plugin so far.

However, you should update your twig templates to avoid the deprecation warnings as shown here https://github.com/cakephp/twig-view#accessing-view.

From:
`{%- element 'Controller/' ~ action %}`

To:
`{{ element('Controller/' ~ action) }}`

**Comment by othercorey** [comment]
Can you add logging to confirm this event handler is run?

https://github.com/geoidesic/CakeQuasarAdmin/blob/500bc084eedfe6e0d6051968c95f6228ca16df55/src/Plugin.php#L29

**Comment by geoidesic** [comment]
Oh hey, somehow I haven't been getting notifications about this. I'll have a look.

**Comment by geoidesic** [comment]
> @geoidesic I am looking at this again. I want to check the issue is still that the custom QuasarAdmin.Bake helper seem to not be used? We talked a lot about loading and twig porting.

@othercorey So in answer. Yes, the issue is still that the custom QuasarAdmin.Bake helper is not being used. The reason we talked about loading and twig porting is because the problem occurred when the twig library was changed, so I assumed it has something to do with that... and so I shared all the information about twig that I could find.

**Comment by geoidesic** [comment]
> Can you add logging to confirm this event handler is run?
> 
> https://github.com/geoidesic/CakeQuasarAdmin/blob/500bc084eedfe6e0d6051968c95f6228ca16df55/src/Plugin.php#L29

@othercorey not too sure what you mean... do you want me to 
A) add in logging for this to my repo so that you can use it, or are you asking me to:
B) test on the installation where the problem exists whether that event handler is being run?

**Comment by othercorey** [comment]
> @othercorey not too sure what you mean... do you want me to
> A) add in logging for this to my repo so that you can use it, or are you asking me to:
> B) test on the installation where the problem exists whether that event handler is being run?
> 
> I assume you mean B. I'll try get to this next week.

Yes, B. Thanks!

**Comment by geoidesic** [comment]
@othercorey ok so I had a go at this but I quickly realised that I have no idea how to add logging in the bake context.
I also wasn't clear on whether you needed me to test this on the old (working) installation or the new (not working) installation.

So I tried putting a `die()` statement into the event handler, that is triggered, like so... 
```
public function bootstrap(PluginApplicationInterface $app): void
    {
        EventManager::instance()->on('Bake.initialize', function (Event $event) {
            $view = $event->getSubject();
            die('happy');
            //- unregister the standard bake helper and register the Quasar helper instead
            $view->helpers()->unload('Bake')->load('QuasarAdmin.Bake');
        });
    }
```

I did that on both versions (i.e. old working, new not-working). On the old-working version the die statement was triggered. On the new not-working version the die statement was not triggered. Conclusion – that event isn't being called. I also tried putting the die statement outside the event handler within the bootstrap method... that also wasn't triggered. 

So it looks like the plugin bootstrap is not being called at all.
